src/Pact/Native.hs

```diff
@@ -155,6 +155,19 @@ enforceOneDef =
           Just b' -> return b'
     enforceOne i as = argsError' i as
 
+tryDef :: NativeDef
+tryDef =
+  defNative "try" try' (funType a [("default", a), ("action", a)])
+  ["(try 3 (enforce (= 1 2) \"this will definitely fail\"))"]
+  "Attempt a pure ACTION, returning DEFAULT in the case of failure"
+  where
+    try' :: NativeFun e
+    try' i as@[da, action] = runReadOnly i $ gasUnreduced i as $ do
+      ra <- reduce da
```

Comment:
This is separate because we _want_ to reduce the default argument to ensure it succeeds, catching only on the action